### DISCUSSION TITLE
Foonathan Memory error when compiling Fast-DDS for QNX SDP 7.1 [13680]

### DISCUSSION THREAD

**Discussion by aniketshukla1** [eProsima/Fast-DDS#2543] [Unofficial support]
Hi Everyone,

I'm cross compiling fast-dds for QNX platform. I'm facing foonathan_memory issue. I'm attaching the error and the toolchain file as well  for the reference.

[error.docx](https://github.com/eProsima/Fast-DDS/files/6958834/error.docx)
[fastdds_qnx.zip](https://github.com/eProsima/Fast-DDS/files/6958760/fastdds_qnx.zip)

Steps followed for building :- 

fastdds version - v2.3.3
foonathan_memory version - latest
fastcdr vesrion - latest

git clone https://github.com/eProsima/Fast-DDS.git
cd Fast-DDS
git checkout v2.3.3
mkdir build -p && cd build
source ~/qnx700/qnxsdp-env.sh
cmake .. -DCMAKE_TOOLCHAIN_FILE=qnx_toolchain.cmake -DCOMPILE_EXAMPLES=OFF -DCMAKE_SHARED_LINKER_FLAGS=-lsocket -DCMAKE_CXX_FLAGS=-isystem\ /my/absolute/path/to/folder/containing/__config
make -j3
make install

Can anyone suggest what mistake I'm doing in these steps. Thanks.

**Comment by EduPonz** [comment]
Hi @aniketshukla619 ,

QNX has offical documentation on how to cross-compile ROS 2 [here](https://ros2-qnx-documentation.readthedocs.io/en/latest/). You could modify those scripts a bit to just cross-compile Fast DDS and its dependencies, it should not be much problematic.

**Comment by aniketshukla1** [comment]
Hi @EduPonz ,

First of all thanks for the reply. I replicated the steps and was able to build fastdds. As I am new to QNX I'm not able to link the .so file. Could you please help me on this how to link .so file on QNX? That would be a great help. Thanks.

**Comment by JLBuenoLopez** [comment]
I am going to move this issue to the Unofficial support discussion forum. QNX is not an [officially supported platform](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md#platform-support). According to the [CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#unofficial-support), the issues related with non-officially supported platforms should be opened in this forum.
